scripts/run_scrnaseq.R

```diff
@@ -0,0 +1,382 @@
+                                                     "percentage_gene_gc_content", "gene_biotype", "strand", "description"))
+  param[["mt"]] = "^MT-"
+  param[["enrichr_dbs"]] = c("GO_Biological_Process_2018", "KEGG_2019_Human", "WikiPathways_2019_Human")
+}
+
+if (opt[["mouse"]]) {
+  param[["mart_dataset"]] = "mmusculus_gene_ensembl"
+  param[["annot_version"]] = 103
+  param[["annot_main"]] = c(ensembl="ensembl_gene_id", symbol="external_gene_name", entrez="entrezgene_accession")
+  param[["mart_attributes"]] = c(param$annot_main, c("chromosome_name", "start_position", "end_position",
+                                                     "percentage_gene_gc_content", "gene_biotype", "strand", "description"))
+  param[["mt"]] = "^mt-"
+  param[["enrichr_dbs"]] = c("GO_Biological_Process_2018", "KEGG_2019_Mouse", "WikiPathways_2019_Mouse")
+}
+
+
+# Write project and sample description
+# Param project id
+param[["project_id"]] = as.character(opt[["project_id"]])
+
+# Param path_data
+# Read path_data csv file and check values
+if (!file.exists(opt[["path_data_csv"]]))
+  stop("Path to a path data definition file specified via --path-data-csv does not exist or is not a file!")
+param[["path_data"]] = read.csv(normalizePath(opt[["path_data_csv"]]), header = TRUE, sep = ",", quote = "\"",
+                                dec = ".", fill = TRUE, comment.char = "")
+
+if (sum(!names(param$path_data) %in% c("name", "type", "path", "stats")) > 0)
+  stop("Need to provide all columns (name, type, path, stats) in path data definition file specified via --path-data-csv!")
+
+if (sum(!param$path_data$type %in% c("10x","smartseq2")) > 0)
+  stop("Need to provide valid types ('10x' or 'smartseq2') in path-data-csv!")
+
+if (sum(!file.exists(param$path_data$path)) > 0) 
+  stop("At least one path to input data specified in --path-data-csv does not exist!")
+
+if (sum(!is.na(param$path_data$stats)) > 0) {
+  if (sum(!file.exists(param$path_data$stats[!is.na(param$path_data$stats)])) > 0) 
+    stop("At least one path to a metrics_summary.csv file specified in --path-data-csv does not exist!")
+}
+
+# Param downsample_cells_n
+param[["downsample_cells_n"]] = opt[["downsample_cells_n"]]
+
+# Param path_out
+param[["path_out"]] = normalizePath(opt[["path_out"]])
+
+# Change additional parameters individually
+# Param file_known_markers
+if (!is.null(opt[["file_known_markers"]])) {
+  if (!file.exists(opt[["file_known_markers"]])) 
+    stop("Path to a known markers file specified via --file-known-markers does not exist or is not a file!")
+  param[["file_known_markers"]] = normalizePath(opt[["file_known_markers"]])
+} else {
+  param[["file_known_markers"]] = NULL
+}
+
+# Param mart_dataset
+param[["mart_dataset"]] = opt[["mart_dataset"]]
+
+# Param annot_version
+param[["annot_version"]] = opt[["annot_version"]]
+
+# Param annot_main
+param[["annot_main"]] = trimws(unlist(strsplit(trimws(unlist(strsplit(opt[["annot_main"]], ","))), "=")))[c(F,T)]
```

Comment:
Can you please use the magrittr operator so that it is better readable?

```
opt["annot_main"] %>% 
  strsplit(",") %>%
  unlist() %>% ...
```

Reply:
Changed the script accordingly and tried to use as much %>% as possible.